rclcpp/include/rclcpp/generic_publisher.hpp

```diff
@@ -78,38 +78,12 @@ class GenericPublisher : public rclcpp::PublisherBase
       node_base,
       topic_name,
       *rclcpp::get_typesupport_handle(topic_type, "rosidl_typesupport_cpp", *ts_lib),
-      options.template to_rcl_publisher_options<rclcpp::SerializedMessage>(qos)),
+      options.template to_rcl_publisher_options<rclcpp::SerializedMessage>(qos),
+      // NOTE(methylDragon): Passing these args separately is necessary for event binding
+      options.event_callbacks,
+      options.use_default_callbacks),
```

Comment:
directly off topic from this PR. but this `use_default_callbacks` flag should be declared in `PublisherEventCallbacks` and `SubscriptionEventCallbacks`? if so, we can just pass `options.event_callbacks` here?

as `PublisherBase` argument, this seems to be more general flag to use default callback for anything.

Reply:
I think this one is kinda tough, because `use_default_callbacks` seems better fit to be part of the options struct than the event_callbacks struct, since it's a configuration for the pub/sub.

I'd have passed the entire options struct in, but that requires templating the base class, which is why I had to go with the compromise of just passing the event callbacks and the bool flag separately...

Reply:
> because use_default_callbacks seems better fit to be part of the options struct than the event_callbacks struct, since it's a configuration for the pub/sub.

i think this is an option for `event_callbacks`, not entire publisher or subscription. that is why i came up with this comment. it would be better if we name this option into `use_default_event_callbacks`?